### PR TITLE
Add slacking for failed builds

### PR DESCRIPTION
**Pull request by mnokka-unikie**
Slacks failed builds to defined channels. Currently dev and priv Jenkins have dedicated Slack channels defined for usage.

### CODE REVIEW

File: ghaf-main-pipeline.groovy [hunk 1]
```diff
@@ -39,8 +39,26 @@ pipeline {
           script {
             env.TARGET_COMMIT = sh(script: 'git rev-parse HEAD', returnStdout: true).trim()
             env.ARTIFACTS_REMOTE_PATH = "${env.JOB_NAME}/build_${env.BUILD_ID}-commit_${env.TARGET_COMMIT}"
+            env.GITHUBLINK="https://github.com/tiiuae/ghaf/commit/${env.TARGET_COMMIT}"
+            env.SERVER_NAME = sh(script: 'echo ${JENKINS_URL}', returnStdout: true).trim()
+                    if (env.SERVER_NAME == "") {
```
Comment (mkaapu) [inline ghaf-main-pipeline.groovy:44]: Fix indentation

**Review by mkaapu — CHANGES_REQUESTED** [review]

File: ghaf-main-pipeline.groovy (hunk 1 above)
Comment (henrirosten) [reply ghaf-main-pipeline.groovy:44]: Note: Not only here, but throughout the changes you have made.

File: utils.groovy [hunk 2]
```diff
@@ -152,6 +152,19 @@ def find_img_relpath(String flakeref, String subdir) {
...
+   def PROD_SLACK_CHANNEL = "FALSE" // no slacking
+   def PRIVATE_SLACK_CHANNEL = "ghaf-jenkins-ci"
+    if (serverName.contains('prod')) {
+        return [env: 'prod', channel:PROD_SLACK_CHANNEL ]
+    } else if (serverName.contains('dev')) {
+        return [env: 'dev', channel:DEV_SLACK_CHANNEL ]
+    } else {
+        return [env:'priv', channel:PRIVATE_SLACK_CHANNEL ]
+    }
+}
```
Comment (henrirosten) [inline utils.groovy:166]: There are several problems with this function, most importantly, I don't think we can draw a conclusion that if the server name contains 'dev' or 'prod', that we could generally determine that it's then 'dev' or 'prod' instance.

I would propose the following:
- Completely remove this function
- Start with adding 'slacking' to one pipeline, perhaps to the `ghaf-main-pipeline.groovy`
- In the post step of the given pipeline, determine if 'slacking' is needed based on the hostname (as returned by `uname -n`). Only proceed with 'slacking' if hostname matches `ghaf-jenkins-controller-dev`

**Comment by mnokka-unikie** [comment]
closing, changes implemented top of current main in a new PR

**Comment by henrirosten** [comment]
@mnokka-unikie : generally, you don't need to introduce a new PR to have it rebased on the new main. Simply rebase locally, then force-push to your branch and it will update the PR too.

**Pull request closed without merging**